plugins/lsp/schemastore.nix

```diff
@@ -0,0 +1,151 @@
+            A list of strings representing the names of schemas to select.
+            If this option is not present, all schemas are returned.
+            If it is present, only the selected schemas are returned.
+            `select` and `ignore` are mutually exclusive.
+
+            See the [schema catalog](https://github.com/SchemaStore/schemastore/blob/master/src/api/json/catalog.json).
+          '';
+
+          ignore = helpers.defaultNullOpts.mkListOf types.str "[]" ''
+            A list of strings representing the names of schemas to ignore.
+            `select` and `ignore` are mutually exclusive.
+
+            See the [schema catalog](https://github.com/SchemaStore/schemastore/blob/master/src/api/json/catalog.json).
+          '';
+
+          replace = helpers.defaultNullOpts.mkAttrsOf schemaEntry "{}" ''
+            An attrs of elements representing schemas to replace with a custom schema.
+            The string key is the name of the schema to replace, the table value is the schema definition.
+            If a schema with the given name isn't found, the custom schema will not be returned.
+          '';
+
+          extra = helpers.defaultNullOpts.mkListOf schemaEntry "[]" ''
+            Additional schemas to include.
+          '';
+        };
+    in
+    {
+      json = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = "Whether to enable the json schemas in jsonls.";
+        };
+
+        settings = helpers.mkSettingsOption {
+          options = schemaOpts;
+          description = "Options supplied to the `require('schemastore').json.schemas` function.";
+          example = {
+            replace."package.json" = {
+              description = "package.json overridden";
+              fileMatch = [ "package.json" ];
+              name = "package.json";
+              url = "https://example.com/package.json";
+            };
+            extra = [
+              {
+                description = "My custom JSON schema";
+                fileMatch = "foo.json";
+                name = "foo.json";
+                url = "https://example.com/schema/foo.json";
+              }
+              {
+                description = "My other custom JSON schema";
+                fileMatch = [
+                  "bar.json"
+                  ".baar.json"
+                ];
+                name = "bar.json";
+                url = "https://example.com/schema/bar.json";
+              }
+            ];
+          };
+        };
+      };
+
+      yaml = {
+        enable = mkOption {
+          type = types.bool;
+          default = true;
+          example = false;
+          description = "Whether to enable the yaml schemas in yamlls.";
+        };
+
+        settings = helpers.mkSettingsOption {
+          options = schemaOpts;
+          description = "Options supplied to the `require('schemastore').yaml.schemas` function.";
```

Comment:
Is there a reason the json settings option has an example but yaml does not? Upstream docs lacking an example?

Reply:
Yes, lack of example upstream. Also, I think it is fine not to put examples absolutely everywhere.